### PR TITLE
Bare Metal Forms and Helpers

### PR DESCRIPTION
**Pull request by craigs40**
This project includes multiple ways to build a form in rails that can send parameters to a controller so that it can build an instance of a model with those parameters.

### CODE REVIEW

**Review by BrianSammit — COMMENTED** [review]
STATUS INVALID :yellow_circle: 
------------
Friendly reminder: You should make sure that your repo looks professional before review.
Remember that you must have a readme file. Also, check the correct use of Git Flow you are missing the main/master branch. 

Please make these required changes and send it again for another review. 

tenor (1)

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS: INVALID 🔴 
### @craigs40, your root#index is not available in the controller kindly fix all requirement and push for review  **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/create_new_user.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
Screenshot 2021-05-26 181224

File: README.md
Comment (ElijahTobs) [inline README.md:3]: - [x] *[Required]* You should add instructions on how to run your project

File: README.md
Comment (ElijahTobs) [inline README.md]: - [x] *[Required]* This currently returns a 404 error. Kindly add the right link

File: config/routes.rb
Comment (ElijahTobs) [inline config/routes.rb]: - [x] *[Required]* This should be:
`resources :users, only: [:new, :create, :edit, :update]`

File: app/controllers/users_controller.rb
Comment (ElijahTobs) [inline app/controllers/users_controller.rb:19]: - [x] *[Required]* Your `UsersController` is missing the `update` action

File: app/views/users/new.html.erb
Comment (ElijahTobs) [inline app/views/users/new.html.erb]: - [x] *[Required]* This should use `'users_path'` for action instead of `'/users'`

File: app/views/users/new.html.erb
Comment (ElijahTobs) [inline app/views/users/new.html.erb]: - [x] *[Required]* This part of the commented out form should use `user[<field_name>]` for **input's name**. For `username`, it will be `user[username]`

File: app/views/users/new.html.erb
Comment (ElijahTobs) [inline app/views/users/new.html.erb]: - [x] *[Required]* The `text_field_tag` here should be `email_field_tag`

File: app/views/users/new.html.erb
Comment (ElijahTobs) [inline app/views/users/new.html.erb]: - [x] *[Required]* The action here should also be `'users_path'`, not `'/users'`

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required ⚠️ 🚦
## Hi @craigs40  👋
You've done really well so far! 👍 
You've followed the Git flow rules. ✔️ 
You have good commit messages ✔️ 

However, there are some changes that you are required to make before this project can be approved.

## Required Changes ♻️
**Note: This set of changes are marked as Required.**

_See the inline comments below_

## Optional Changes 👌
_Every comment with the OPTIONAL prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Check for changes marked as Optional in the inline comments below.


#### If anything doesn't seem clear to you, feel free to leave your question or comment in the PR thread below

## Happy Coding :computer: 

_Reviewed by_  **Elijah O. Ayandokun**

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @craigs40! :wave: :wave:

####  Good job for work making all the necessary corrections recommended by the previous reviewer.  👍  Congratulations.  :clap:


#### You may merge, good luck on your next project! :rocket:

If you have any questions pls feel free to drop me a comment.

#### Happy Coding! :+1:

__Robert Loterh __
[GitHub](https://github.com/rloterh) | [LinkedIn](https://www.linkedin.com/in/rloterh/) | [Twitter](https://twitter.com/RLoterh)

**Pull request merged**
